### PR TITLE
MAT-6654 fixed fhir terminology validation

### PR DESCRIPTION
**Pull request by sb-prateekkeerthi**
Screenshot 2024-02-02 at 1 12 25 PM
Screenshot 2024-02-02 at 10 19 55 AM

## MADiE PR

Jira Ticket: [MAT-6654](https://jira.cms.gov/browse/MAT-6654)
(Optional) Related Tickets:

### Summary
fixed fhir terminology validation

### All Submissions
* [ ] This PR has the JIRA linked.
* [ ] Required tests are included
* [ ] No extemporaneous files are included (i.e Complied files or testing results)
* [ ] This PR is in to the **correct branch**.
* [ ] All Documentation as needed for this PR is Complete (or noted in a TODO or other Ticket)
* [ ] Any breaking changes or failing automation are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [ ] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package)
* [ ] All CDN/Web dependancies are hosted internally (i.e MADiE-Root Repo)

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
*  The tests appropriately test the new code, including edge cases
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads

### CODE REVIEW

File: src/validations/valuesetValidation.ts [hunk 1]
```diff
@@ -15,6 +15,20 @@ const GetValueSetErrors = async (
...
+          if (
+            valuesetUrlWithoutOid &&
+            valuesetUrlWithoutOid[0] !== "'http://cts.nlm.nih.gov/fhir/"
+          ) {
+            const invalidValuesetUrlError =
+              processValueSetErrorForElmTranslationError(
+                `"${valueSet.url} is not a valid URL"`,
+                locator
+              );
+            valuesetsErrorArray.push(invalidValuesetUrlError);
```
Comment (RohitKandimalla) [inline src/validations/valuesetValidation.ts]: if a URL is not valid, should we skip the call to `terminologyServiceApi.getValueSet()` ?

File: src/validations/valuesetValidation.ts [hunk 2]
```diff
@@ -15,6 +15,20 @@ const GetValueSetErrors = async (
...
         const locator = getLocatorFromCqlValueSet(valueSet);
+        if (model === "QICore") {
+          const valuesetUrlWithoutOid = valueSet?.url?.split("ValueSet/");
+          if (
+            valuesetUrlWithoutOid &&
+            valuesetUrlWithoutOid[0] !== "'http://cts.nlm.nih.gov/fhir/"
+          ) {
+            const invalidValuesetUrlError =
+              processValueSetErrorForElmTranslationError(
+                `"${valueSet.url} is not a valid URL"`,
```
Comment (RohitKandimalla) [inline src/validations/valuesetValidation.ts]: Do we want to enhance the error message suggesting what is the correct URL format? I think we need to ask POs or Michael. I don't see a down side by telling what is the accepted URL.

File: src/validations/valuesetValidation.ts (hunk 1 above)
Comment (RohitKandimalla) [reply src/validations/valuesetValidation.ts]: It looks like we could first perform checks to make sure the ValueSet is valid and if valid, we can proceed to get OID , Locator and make a call to API 
If not a valid URL, then skip the steps

Comment (sb-prateekkeerthi) [reply src/validations/valuesetValidation.ts]: No this would be seperate, because terminologyServiceApi.getValueSet() is concerned only with oid check whereas the above code is concerned about the url not the oid. suppose consider a situation where both url and oid are not valid ones in that case we need to provide 2 validation errors one for oid and other for the url.

Comment (RohitKandimalla) [reply src/validations/valuesetValidation.ts]: Do we display both these errors in the gutter on the same line ? Can you add screenshots to this PR if possible ?

File: src/validations/valuesetValidation.ts (hunk 2 above)
Comment (sb-prateekkeerthi) [reply src/validations/valuesetValidation.ts]: thanks for the suggestion, I have updated the message based on the POs suggestion

**Review by RohitKandimalla — APPROVED** [review]
